File: ePrijevozSarajevo/ePrijevozSarajevo.Services/Database/IssuedTicket.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ePrijevozSarajevo.Services.Database
{
    public class IssuedTicket
    {
        public int IssuedTicketId { get; set; }
        public User? User { get; set; }
        public int? UserId { get; set; }
        public Ticket? Ticket { get; set; } 
        public int? TicketId { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime ValidTo { get; set; }
        public DateTime IssuedDate { get; set; } = DateTime.UtcNow;
        public int? Amount { get; set; }
        public Route? Route { get; set; }
        public int? RouteId { get; set; }
    }
}

File: ePrijevozSarajevo/UI/eprijevoz_mobile/lib/models/issuedTicket.dart
// import 'package:json_annotation/json_annotation.dart';
// part 'issuedTicket.g.dart';

// @JsonSerializable()
// class IssuedTicket {
//   int? issuedTicketId;
//   int? userId;
//   int? ticketId;
//   DateTime? validFrom;
//   DateTime? validTo;
//   DateTime? issuedDate;

//   IssuedTicket({
//     this.issuedTicketId,
//     this.userId,
//     this.ticketId,
//     this.validFrom,
//     this.validTo,
//     this.issuedDate,
//   });

//   factory IssuedTicket.fromJson(Map<String, dynamic> json) =>
//       _$IssuedTicketFromJson(json);

//   Map<String, dynamic> toJson() => _$IssuedTicketToJson(this);
// }
import 'package:json_annotation/json_annotation.dart';
part 'issuedTicket.g.dart';

@JsonSerializable()
class IssuedTicket {
  int? issuedTicketId;
  int? userId;
  int? ticketId;
  DateTime? validFrom;
  DateTime? validTo;
  DateTime? issuedDate;

  IssuedTicket({
    this.issuedTicketId,
    this.userId,
    this.ticketId,
    this.validFrom,
    this.validTo,
    this.issuedDate,
  });

  factory IssuedTicket.fromJson(Map<String, dynamic> json) =>
      _$IssuedTicketFromJson(json);

  Map<String, dynamic> toJson() => _$IssuedTicketToJson(this);
}
